javatests/com/williamfiset/algorithms/strings/BoyerMooreStringSearchTest.java

```diff
@@ -0,0 +1,46 @@
+package javatests.com.williamfiset.algorithms.strings;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.williamfiset.algorithms.strings.BoyerMooreStringSearch;
+import org.junit.Before;
+import org.junit.Test;
+
+public class BoyerMooreStringSearchTest {
+
+  private BoyerMooreStringSearch underTest;
+
+  @Before
+  public void setup() {
+    underTest = new BoyerMooreStringSearch();
+  }
+
+  @Test
+  public void shouldReturnEmptyListOnNullInput() {
+    assertThat(underTest.findOccurrences(null, "")).isEmpty();
+    assertThat(underTest.findOccurrences("", null)).isEmpty();
+    assertThat(underTest.findOccurrences(null, null)).isEmpty();
+  }
+
```

Comment:
Request: Can you add a few more tests? What is the expected behavior for an empty text? What about an empty pattern? A pattern longer than the text? No matches at all? etc..